packages/ai-chat-ui/src/browser/ai-chat-ui-contribution.ts

```diff
@@ -83,10 +84,21 @@ export class AIChatContribution extends AbstractViewContribution<ChatViewWidget>
             })
         });
         registry.registerCommand(AI_CHAT_NEW_CHAT_WINDOW_COMMAND, {
-            execute: () => this.chatService.createSession(ChatAgentLocation.Panel, { focus: true }),
-            isEnabled: widget => this.withWidget(widget, () => true),
+            execute: () => this.openView().then(() => this.chatService.createSession(ChatAgentLocation.Panel, { focus: true })),
             isVisible: widget => this.withWidget(widget, () => true),
         });
+        registry.registerCommand(AI_CHAT_NEW_WITH_MEMORY, {
+            execute: () => {
+                const activeSessions = this.chatService.getSessions().filter(candidate => candidate.isActive);
+                if (activeSessions.length !== 1) {
+                    return;
+                }
+                const activeSession = activeSessions[0];
+                const newSession = this.chatService.createSession(ChatAgentLocation.Panel, { focus: true }, activeSession.pinnedAgent);
+                newSession.model.context.addVariables({ variable: SESSION_SUMMARY_VARIABLE, arg: activeSession.id });
```

Comment:
I just tried out to adopt this feature in another agent which leads again to the issue that the actual context creation happens inside of the resolvement. I would make the variable "stupid". Caller have to create the context themselve using the service. Otherwise you can for example as an agent creating a context not influence the prompt, although the task service would allow to do this. So the variable would only get the id of the task context and this task context must exist, otherwise it will just not resolve.

I would expect this here to be like:
id = task-context-service.summarizeAndStore(activeSession)
newSession.model.context.addVariables({ variable: SESSION_SUMMARY_VARIABLE, arg: id });

Does this make sense?